### PR TITLE
bpo-46745: Fix typo in PositionsIterator

### PR DESCRIPTION
**Pull request by eclipseo**
https://bugs.python.org/issue46745

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@eclipseo

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Objects/codeobject.c [hunk 1]
```diff
@@ -1020,7 +1020,7 @@ positionsiter_next(positionsiterator* pi)
 
 static PyTypeObject PositionsIterator = {
     PyVarObject_HEAD_INIT(&PyType_Type, 0)
-    "poisitions_iterator",              /* tp_name */
+    "positions_iterator",              /* tp_name */
```
Comment (JelleZijlstra) [inline Objects/codeobject.c]:
```suggestion
    "positions_iterator",               /* tp_name */
```
Keep the comment aligned

**Review by JelleZijlstra — APPROVED** [review]
Thanks! I'll merge this PR in a few days (cc @gvanrossum).

**Review by gvanrossum — APPROVED** [review]
No need to wait for this.

**Comment by JelleZijlstra** [comment]
Yeah I felt like just merging this already would be fine :)

**Pull request merged**